dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship/Functions/GetApprenticeshipsAsProvider.cs

```diff
@@ -39,12 +50,54 @@ public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous,
                     return new EmptyResult();
                 }
 
-                var providers = _cache.GetOrAdd("DasProviders", () => _apprenticeshipService.ApprenticeshipsToDasProviders(persisted), DateTimeOffset.Now.AddHours(8));
-                
-                return new OkObjectResult(providers);
+                var result = await _cache.GetOrAddAsync<(string ExportKey, string Value)>("DasProviders", async () =>
```

Comment:
The addition of this code makes this function definition rather long, could you extract some of it into private methods or something?

Reply:
Agreed.